File: src/VSP/Commands/VsCommands.cs
namespace VSP.Commands
{
    public class VsCommands
    {
        private readonly VsHelper vsHelper;

        public VsCommands(VsHelper vsHelper)
        {
            this.vsHelper = vsHelper;
        }

        /// <summary>
        /// Executes a Visual Studio command
        /// </summary>
        /// <param name="commandName">command name</param>
        /// <param name="commandArgs">command args</param>
        /// <example>
        /// Opens a File.NewFile dialog
        /// vsHelpers.Commands.ExecuteCommand("File.NewFile")
        /// </example>
        public void Execute(string commandName, string commandArgs)
        {
            this.vsHelper.DTE.ExecuteCommand(commandName, commandArgs);
        }

        public EnvDTE.Command Get(string commandName, int id)
        {
            return this.vsHelper.DTE.Commands.Item(commandName, id);
        }

    }
}

File: src/VsErc/scripts/commands/vs/vscommands.lua
local com = require('com')

erc.editor.vs.commands = {}

erc.editor.vs.commands.execute = function(commandname, commandsargs)
	com.wrap(erc.editor.vs.dte()):ExecuteCommand(commandname, commandsargs or '')
end

erc.editor.vs.commands.get = function (commandname, commandsargs)
	return erc.editor.vs._vshelper.Commands:Get(commandname, commandsargs or -1)
end